### ISSUE TITLE
Card layout constraint issue [Help Wanted]

### ISSUE THREAD

**Issue by HybridHB** [CosmicMind/Material#883]
Hello, Im having issues with a scroll view with 2 cardViews as subviews. The card views are arranged one above another (cardA above cardB). They are arranged on the screen using the layout frame work.

scrollView.addSubview(cardA)
scrollView.addSubview(cardB)
scrollView.layout(cardA).horizontally(left: 8, right: 8).centerHorizontally().top(8)
scrollView.layout(cardB).horizontally(left: 8, right: 8).centerHorizontally().top(cardA.frame.height + 16)

The user is able to interact with other items on the view during runtime which can change the height of cardB.contentView. When cardB.setNeedsLayout() is called, cardB resizes itself and everything looks correct on the card itself. The issue is that the top and bottom margin of cardB changes during the resize and messes up the spacing of cardB relative to other views. Its like cardB is anchored from the center. Is there a way to get cardB to resize while keeping the same spacing between cardA and cardB? Any help would be appreciated, thanks!

**Comment by danieldahan** [comment]
Hey! Have you had a chance to look at these two samples, [CardTableView](https://github.com/CosmicMind/Samples/tree/master/Projects/Programmatic/CardTableView) and [CollectionView](https://github.com/CosmicMind/Samples/tree/master/Projects/Programmatic/CollectionView)?

One uses a CollectionView and the other a TableView to accomplish a feed style view controller with Cards. That said, is that what you are looking to accomplish? If not, can you show me an image of what you want to accomplish, and explain how these cards are being updated? Thank you!

**Comment by danieldahan** [comment]
I haven't heard from you. If you still face the issue, please reopen it. Thank you!

**Comment by HybridHB** [comment]
Sorry for the late response, Ive been working on converting my view to use a CollectionView as you suggested and finally got it working. FWIW my original code used to work on release 2.4.12. I only recently updated the material package for swift 4 support and thats when I saw the issue of xy positions changing after a card size change and layout although I dont know when exactly layout changed in behavior. Anyways, thanks for the help!
